DEVELOPING.md

```diff
@@ -77,134 +51,23 @@ Alternately you can invoke ruff by typing the following from within the `./pytho
-  <!-- markdown-link-check-enable -->
-  To create a new documentation page, add a markdown file under `/docs/` and add a reference to this page in `mkdocs.yml` under `nav`.
-
-### <a name="api"></a>Updating API documentation
-
-If you've made changes to the core `uagents` package that affects the public API:
-
-- ``` shell
-   make generate-api-docs
-  ```
-
-  This regenerates the API docs. If pages are added/deleted, or there are changes in their structure, these need to be reflected manually in the `nav` section of `mkdocs.yaml`.
-
-### <a name="deps"></a>Updating dependencies
-
-We use [`poetry`][poetry] and `pyproject.toml` to manage the project's dependencies.
-
-If you've made any changes to the dependencies (e.g. added/removed dependencies, or updated package version requirements):
-
-- ``` shell
-   poetry lock
-  ```
-
-  This re-locks the dependencies. Ensure that the `poetry.lock` file is pushed into the repository (by default it is).
-
-  Note: Whenever the lock file is updated, you need to make sure the version of `tox` installed in the CI workflows matches the version in `poetry.lock`.
-
-- ``` shell
-   make liccheck
-  ```
-
-  Checks that the licence for the library is correct, taking into account the licences for all dependencies, their dependencies and so forth.
-
-### <a name="tests"></a>Tests
-
-To test the project, we use `pytest`. To run the tests:
-
-- ``` shell
-   make test
-  ```
-
-  Runs all the tests.
-
-- ``` shell
-   make unit-test
-   ```
-
-  Runs all unit tests.
-
-- ``` shell
-   make integration-test
-  ```
-
-  Runs all integration tests.
-
-- ``` shell
-   make coverage-report
-  ```
-
-  Produces a coverage report (you should run tests using one of the above commands first).
-
-### <a name="misc"></a>Miscellaneous checks
-
-- ``` shell
-   make copyright-check
-  ```
-
-  Checks that all files have the correct copyright header (where applicable).
-
 ## <a name="contributing"></a>Contributing
 
 <!-- markdown-link-check-disable -->
-For instructions on how to contribute to the project (e.g. creating Pull Requests, commit message convention), see the [contributing guide](CONTRIBUTING.md).
+For instructions on how to contribute to the project (e.g. creating Pull Requests, commit message convention, ...), see the [contributing guide](CONTRIBUTING.md).
 <!-- markdown-link-check-enable -->
 
-[mkdocs]: https://www.mkdocs.org
-[material]: https://squidfunk.github.io/mkdocs-material/
-[poetry]: https://python-poetry.org
-[contributing guide]: https://github.com/fetchai/uAgents/blob/main/CONTRIBUTING.md
-[repo]: https://github.com/fetchai/uAgents
+- [mkdocs](https://www.mkdocs.org)
+- [material](https://squidfunk.github.io/mkdocs-material/)
+- [poetry](https://python-poetry.org)
+- [contributing guide](CONTRIBUTING.md)
+- [repo](https://github.com/fetchai/uAgents)
```

Comment:
These are references for links used across the file. 
- Remove the ones not used
- Revert back the `:`
- use poetry and repo if possible otherwise remove